src/Components/Blazor/WebAssembly.Authentication/src/WebAssemblyAuthenticationServiceCollectionExtensions.cs

```diff
@@ -0,0 +1,126 @@
+        /// <typeparam name="TProviderOptions">The configuration options of the underlying provider being used for handling the authentication operations.</typeparam>
+        /// <param name="services">The <see cref="IServiceCollection"/> to add the services to.</param>
+        /// <param name="configure">An action that will configure the <see cref="RemoteAuthenticationOptions{TProviderOptions}"/>.</param>
+        /// <returns>The <see cref="IServiceCollection"/> where the services were registered.</returns>
+        public static IServiceCollection AddSpaAuthentication<TRemoteAuthenticationState, TProviderOptions>(this IServiceCollection services, Action<RemoteAuthenticationOptions<TProviderOptions>> configure)
+            where TRemoteAuthenticationState : RemoteAuthenticationState
+            where TProviderOptions : new()
+        {
+            services.AddSpaAuthentication<RemoteAuthenticationState, TProviderOptions>();
+            if (configure != null)
+            {
+                services.Configure(configure);
+            }
+
+            return services;
+        }
+
+        /// <summary>
+        /// Adds support for authentication for SPA applications using <see cref="OidcProviderOptions"/> and the <see cref="RemoteAuthenticationState"/>.
+        /// </summary>
+        /// <param name="services">The <see cref="IServiceCollection"/> to add the services to.</param>
+        /// <param name="configure">An action that will configure the <see cref="RemoteAuthenticationOptions{TProviderOptions}"/>.</param>
+        /// <returns>The <see cref="IServiceCollection"/> where the services were registered.</returns>
+        public static IServiceCollection AddOidcAuthentication(this IServiceCollection services, Action<RemoteAuthenticationOptions<OidcProviderOptions>> configure)
+        {
+            AddSpaAuthentication<RemoteAuthenticationState, OidcProviderOptions>(services, configure);
+
+            services.TryAddEnumerable(ServiceDescriptor.Singleton<IPostConfigureOptions<RemoteAuthenticationOptions<OidcProviderOptions>>, DefaultOidcOptionsConfiguration>());
+
+            if (configure != null)
+            {
+                services.Configure(configure);
+            }
+
+            return services;
+        }
+
+        /// <summary>
+        /// Adds support for authentication for SPA applications using <see cref="ApiAuthorizationProviderOptions"/> and the <see cref="RemoteAuthenticationState"/>.
+        /// </summary>
+        /// <param name="services">The <see cref="IServiceCollection"/> to add the services to.</param>
+        /// <returns>The <see cref="IServiceCollection"/> where the services were registered.</returns>
+        public static IServiceCollection AddApiAuthorization(this IServiceCollection services)
+        {
+            var inferredClientId = Assembly.GetCallingAssembly().GetName().Name;
```

Comment:
Does this client ID have to be "correct" in any sense, or is it just an arbitrary string?

If it's an arbitrary string for info only then great. But if it ever plays a functional role, I suspect we need to get developers to specify it explicitly.